### DISCUSSION TITLE
Create Template

### DISCUSSION THREAD

**Discussion by WR25** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#95] [General]
Hello everyone,

My name is Hamza. I want to express my gratitude for providing such a useful package. I am excited to use it, but I need some assistance. Could anyone provide an example or guidance on how to create a template? I've tried multiple times but haven't achieved a good result.

Thank you very much!

**Comment by gabrieldwight** [comment]
Hi, you can find more information from the documentation

[Template Messages](https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-message-templates/)
[Create message templates for your WhatsApp Business account](https://www.facebook.com/business/help/2055875911147364?id=2129163877102343)

**Reply by WR25** [reply]
thank you so much . but i was talk about create template with api there fuction call  CreateTemplateMessageAsync() when try use it all time give me (#100) The parameter name is required.
 this is method i use:
 string templateName = """
                    {
      "name": "seasonal_promotion",
      "language": "en_US",
      "category": "UTILITY",
      "components": [
        
        {
          "type": "BODY",
          "text": "Shop now through  and use codeto get  off of all merchandise."
          
        }
        
        
      ]
    }
    """;

try
{

    var result = await _metaService._businessClient.CreateTemplateMessageAsync("116289968120746", templateName);


    return Ok(result);
}
catch (WhatsappBusinessCloudAPIException ex)
{
    return BadRequest(ex.Message);
}

**Reply by gabrieldwight** [reply]
you will have to construct the template message using anonymous type then send it as an object in the createtemplateasync function.

[Anonymous Types](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/anonymous-types)

**Reply by WR25** [reply]
Thank you for help is work now . thank you
